### ISSUE TITLE
Outpost management perks stop working

### ISSUE THREAD

**Issue by Constellation-VASCO** [Starfield-Community-Patch/Starfield-Community-Patch#419]
I noticed this by chance. In my first playthrough I upgraded all outpost related perks to the max (so 6/6 robots, more cargo links, etc.) At some point, after grinding all my powers to level 10 I decided this would be my ultimate playthrough and went ahead to do some serious outpost building to notice that I suddenly could only build 3/3 robots and way less cargo links than I should be able to. I have meanwhile started a new game, and have run into the same issue. 

---

## Details 
I do not know how or when it happened, it appears to be connected to NG+ (though not confirmed), multiple users report this problem. A link to a random reddit thread confirming this is apparently widespread is below.

https://www.reddit.com/r/Starfield/s/7yIpDdVR9G

My programming abilities are too limited for me to able to pin down the issue. All I can say is that the problem exists on both Xbox/MS store as well as Steam versions of the game, as I own both and have new playthroughs on both platforms

## Game Version
1.7.36.0

## Language
English (EN)

## New Game
Yes

## Reproduction Steps
See summary, I do not know when the bug triggered, all I know at this stage is I'm not the only one affected. See a random reddit thread on this issue below. My "gut feeling" is that it started with the new patch, but it could be coincidental that I only noticed it then, took me a while to grind my powers to level 10 after all.

https://www.reddit.com/r/Starfield/s/7yIpDdVR9G

Note: This makes my "building problem" issue redundant, as I felt I had chosen the title poorly

**Comment by Constellation-VASCO** [comment]
I have the same bug.

**Comment by Constellation-VASCO** [comment]
I ran into the exact same situation, almost a rage-quit moment because I was looking forward to building a giant set of bases at NG+10. I had alot of hard-saves, so I could test if going to NG+ had something to do with it. I found the run where I got my first rank of outpost building, and had a save game just prior to going to the next NG+, tried it a few times, but could still build 6 cargo-links in the next NG+. 

The only thing I did different from that old save where it did work, moving on to NG+, was that in my real playthru I continued leveling up outpost-building to level 3 in that playthru, and the next save in my archive was in a new NG+ and did have the bug. 

Though it could be NG+ related, I can atleast confirm that it doesn't always go wrong. I know I have this bug, but from a "good point" forward, I could not reproduce it.

**Comment by Constellation-VASCO** [comment]
hi, found a workaround for this issue!
Maybe this helps also in the solution!

Make sure you have achievments enabler mod installed so you do not get locked out of that before doing below commands.

When jumping into a new ng+:
Console:
player.removeperk 0023826F

check if perk is removed

player.addperk 0023826F

Do this last 4 times for each skill level.

and you will have your 6 cargo links, crew stations, robots etc etc again.
Also works if you already have ouposts which suffer from this.

**Comment by Constellation-VASCO** [comment]
I too have this same problem. I do similar, where I build everything in Game 0, and then revisit and rebuild them all again at NG 10

I build 1 base in each NG, and can confirm that any NG resets the perk so you can only do the baseline Outpost Management perk

**Comment by Constellation-VASCO** [comment]
erik, to your comment
make a text file in your game directory:
Add the following:

player.removeperk 0023826F
player.removeperk 0023826F
player.removeperk 0023826F
player.removeperk 0023826F
player.addperk 0023826F
player.addperk 0023826F
player.addperk 0023826F
player.addperk 0023826F

Save it as "ngoutpostfix.txt"

In game type 

bat ngoutpostfix

and it'll be fixed up

**Comment by Constellation-VASCO** [comment]
Been doing some more investigating, and it appears that the script fragment to increment the outpost values only triggers upon acquiring the skill, and isn't reapplied when you pass through unity.
I don't know how to fix this (yet) but maybe one of y'all much smarter folks than I already know.
Quest ID is "PlayerSkills"
Quest Stages 701 through 704 are Outpost Management Rank 1-4 respectively.

Script Name : Fragments:Quests:QF_PlayerSkills_002C59E4
Function Name : Fragment_Stage_0701_Item_00

Ranks 1-3 have fragments associated.

Basically the master "I just came through unity" script needs to check the skill level, and re-run those script fragments for each rank of outpost Management, and I think it will fix it back to normal operation.

**Comment by Constellation-VASCO** [comment]
Same Issue on 1.13.61.0 with SFCP 0.1.7

**Comment by Deebz96** [comment]
This perk works by increasing the following global values from 3 to 6 at ranks 1-3:

- Outpost_BuildLimit_CargoLinks [GLOB:00190D57]
- Outpost_BuildLimit_Robots [GLOB:001D9F5E]
- Outpost_BuildLimit_CrewStations [GLOB:001B4BB2]

These are three of several global values that are used for various skills. Another one, for example is Skill_Astrophysics_DiscoverTraitChance [GLOB:002508F2], which defaults to 0 and increases as you take the Astrophysics perk.

NG+ does not seem to be the trigger for these values resetting. I have just tested by taking rank 1 of the Outpost Management skill, and jumping through the Unity twice. Outpost_BuildLimit_CargoLinks still correctly returned a 6. Additionally, the Skill_Astrophysics_DiscoverTraitChance global has remained at 20 for me through all 10 NG+ runs I've done on this character (I took two ranks in the first universe, haven't touched it since), and it appears to be set up no differently than the outpost build limit globals.

It would be possible for us to handle this by using an OnPlayerLoadGame event in a player alias script, and checking to see if the global value matches what is expected based on the stages set in the PlayerSkills quest. However, this would very likely be incompatible with mods that change how this skill works, and would not be an ideal solution.

It would be useful to have a better idea of what the actual trigger is. May not be feasible to do a retroactive fix that is compatible with mods, but the core issue of these global values being reset might be possible to address if the cause is better understood.
